exchanges/kucoin/kucoin_websocket.go

```diff
@@ -1686,13 +1683,41 @@ func channelName(s *subscription.Subscription, a asset.Item) string {
 	return s.Channel
 }
 
-// removeSpotFromMargin removes spot pairs from margin pairs in the supplied AssetPairs map for subscriptions to non-margin endpoints
-func removeSpotFromMargin(s *subscription.Subscription, ap map[asset.Item]currency.Pairs, spotPairs currency.Pairs) string {
+// mergeMarginPairs merges margin pairs into spot pairs for shared subs (ticker, orderbook, etc) if Spot asset and sub are enabled,
+// because Kucoin errors on duplicate pairs in separate subs, and doesn't have separate subs for spot and margin
+func (ku *Kucoin) mergeMarginPairs(s *subscription.Subscription, ap map[asset.Item]currency.Pairs) string {
 	if strings.HasPrefix(s.Channel, "/margin") {
 		return ""
 	}
-	if p, ok := ap[asset.Margin]; ok {
-		ap[asset.Margin] = p.Remove(spotPairs...)
+	wantKey := &subscription.IgnoringAssetKey{Subscription: s}
+	switch s.Asset {
+	case asset.All:
+		marginPairs, _ := ku.GetEnabledPairs(asset.Margin)
+		ap[asset.Spot] = common.SortStrings(ap[asset.Spot].Add(marginPairs...))
```

Comment:
Only important for consistency and testing.
They're sorted originally in `fillAssetPairs`. 
Adding to them later implies they should remain sorted.
The reason it matters is because the sequence is what we'll actually use in the QualifiedChannel.
I subjectively wanted that to be consistent and deterministic, but maybe it doesn't have to be.

If we don't sort them here, we'll have to do some magic in testing things like QualifiedChannel batching into sets of 100, etc. Off the top of my head, I think we'd have to lose the test helper for equalSubs if we don't have a deterministic QualifiedChannel, because the helper can't know how to split it, especially for batching pairs.

I'm not advocating runtime performance for testing convenience. The bigger argument for me is subscription message consistency.

Without sorting, this is the failure we get for kucoin:
```
    kucoin_test.go:2295:
                Error Trace:    /Users/gbjk/go/src/github.com/gbjk/gocryptotrader/internal/testing/subscriptions/subscriptions.go:32
                                                        /Users/gbjk/go/src/github.com/gbjk/gocryptotrader/exchanges/kucoin/kucoin_test.go:2295
                Error:          Differences:
                                 + /market/ticker:BTC-USDT,ETH-BTC,ETH-USDT,LTC-USDT,TRX-BTC,SOL-USDC ticker spot BTC/USDT,ETH/BTC,ETH/USDT,LTC/USDT,TRX/BTC,SOL/USDC
                                 + /spotMarket/level2Depth5:BTC-USDT,ETH-BTC,ETH-USDT,LTC-USDT,TRX-BTC,SOL-USDC orderbook spot BTC/USDT,ETH/BTC,ETH/USDT,LTC/USDT,TRX/BTC,SOL/USDC
                                 - /market/ticker:BTC-USDT,ETH-BTC,ETH-USDT,LTC-USDT,SOL-USDC,TRX-BTC ticker spot BTC/USDT,ETH/BTC,ETH/USDT,LTC/USDT,SOL/USDC,TRX/BTC
                                 - /spotMarket/level2Depth5:BTC-USDT,ETH-BTC,ETH-USDT,LTC-USDT,SOL-USDC,TRX-BTC orderbook spot BTC/USDT,ETH/BTC,ETH/USDT,LTC/USDT,SOL/USDC,TRX/BTC
                Test:           TestGenerateSubscriptions
                Messages:       Subscriptions should be equal
```